src/box/iproto.cc

```diff
@@ -118,6 +118,9 @@ struct iproto_thread {
 	 */
```

Comment:
> If connection does not support graceful shutdown, then read data from input socket until iproto packet boundary or until msg_max_limit is reached and shutdown(con, SHUT_RD).

Why do you need to read it? Why can't call shutdown right away? Reading of the pending data won't give any guarantees to the client anyway. You don't even need an explicit shutdown then, just call `close()`. AFAIU, it will do the graceful connection shutdown inside.

> Added extra "return" in iproto_connection_try_to_start_destroy to avoid it.

But you didn't. This commit does not touch iproto_connection_try_to_start_destroy at all.